src/OutputWriters/windowed_time_average.jl

```diff
@@ -204,6 +204,8 @@ function (wta::WindowedTimeAverage)(model)
         model.clock.iteration > 0 &&
         @warn "Returning a WindowedTimeAverage before the collection period is complete."
 
+    stride(wta) > 1 && @warn "WindowedTimeAverage can be erroneous when stride > 1 and either the timestep is variable or there are floating point rounding errors in times, both of which result in a decoupling of the model clock times (used in the OutputWriters) and iteration numbers (used for stride)."
```

Comment:
What does "or there are floating point rounding errors in times" mean?

I think with a warning like this, we need to have an issue that documents the problem and goes into a bit more detail (someone who encounters this might want to fix it)

Reply:
It has to do with errors in the model clock time, as in https://github.com/CliMA/Oceananigans.jl/pull/3606. Because stride relates to the iteration number, it gets thrown off when errors in the model clock force for the model to take extremely small timesteps (even when the model is configured with a constant timestep).

You pointed out yourself that the issue is related to roundoff error in this Issue comment https://github.com/CliMA/Oceananigans.jl/issues/3670#issuecomment-2264548157

This PR fixes that bug for stride=1 by using a similar fix as in https://github.com/CliMA/Oceananigans.jl/pull/3616, but I couldn't figure out how to fix it with stride>1 because stride is inherently tied to iteration number.

Reply:
As an aside, I really don't get the point of supporting `stride>1`. Is anyone actually using this? Is it ever worth trading a slight speed-up to make the time-mean erroneous and inconsistent with model snapshots?

That said, it is possible that https://github.com/CliMA/Oceananigans.jl/pull/3606 would fix this clock-iteration decoupling problem and may make this a moot point.

Reply:
The speed up depends on how expensive the thing is you're trying to compute, so it's not always "slight" (for example, I needed this when I first implemented time averaging for the application I was working on). But still, if having `stride` prevents us from a clean implementation of time averaging, then its fine to discontinue support and view it as a candidate for future performance applications.

Reply:
I have used it a few times. Although I also don't opposite removing it if it's needed to keep a correct implementation simple.

Reply:
Okay, fair enough. Why don't we leave the stride > 1 option in but with a warning that we can't not guarantee its degree of accuracy. Do you have suggestions for how to phrase the warning? I think this is the only requested change holding up this PR!

Reply:
Maybe just restate what the problem is with `stride`? I don't understand why `stride=1` is ok but `stride>1` is not.

Reply:
We can work on the warning in a future PR